src/components/ReportingPayments.js

```diff
@@ -1,11 +1,85 @@
 import { Component, PropTypes, DOM as dom, createFactory } from 'react';
+import _ from 'ramda';
+import $ from '../money';
+import moment from 'moment';
 
 class ReportingPayments extends Component {
+  constructor(props) {
+    super(props);
+    this.transactionsByDate = this.transactionsByDate.bind(this);
+    this.paymentsByDate = this.paymentsByDate.bind(this);
+    this.renderPaymentsByDate = this.renderPaymentsByDate.bind(this);
+    this.renderHeader = this.renderHeader.bind(this);
+  }
+
+  dateRange() {
+    const today = moment().startOf('day');
+    return _.map((days) => moment(today).subtract(days, 'days'), _.reverse(_.range(0, 7)));
+  }
+
```

Comment:
Query db would be passing date range that user selects, and returning transactions for just those, right? As opposed to loading all transaction and filtering by date on the client. If so, yeah I agree.